iac-extensions/arm/src/main/java/org/sonar/iac/arm/visitors/ArmHighlightingVisitor.java

```diff
@@ -0,0 +1,121 @@
+import org.sonar.iac.arm.tree.impl.bicep.AbstractDeclaration;
+import org.sonar.iac.arm.tree.impl.bicep.BooleanLiteralImpl;
+import org.sonar.iac.arm.tree.impl.bicep.DecoratorImpl;
+import org.sonar.iac.arm.tree.impl.bicep.ForExpressionImpl;
+import org.sonar.iac.arm.tree.impl.bicep.ForVariableBlockImpl;
+import org.sonar.iac.arm.tree.impl.bicep.FunctionDeclarationImpl;
+import org.sonar.iac.arm.tree.impl.bicep.IfConditionImpl;
+import org.sonar.iac.arm.tree.impl.bicep.ImportDeclarationImpl;
+import org.sonar.iac.arm.tree.impl.bicep.InterpolatedStringImpl;
+import org.sonar.iac.arm.tree.impl.bicep.ModuleDeclarationImpl;
+import org.sonar.iac.arm.tree.impl.bicep.MultilineStringImpl;
+import org.sonar.iac.arm.tree.impl.bicep.NullLiteralImpl;
+import org.sonar.iac.arm.tree.impl.bicep.NumericLiteralImpl;
+import org.sonar.iac.arm.tree.impl.bicep.OutputDeclarationImpl;
+import org.sonar.iac.arm.tree.impl.bicep.ParameterDeclarationImpl;
+import org.sonar.iac.arm.tree.impl.bicep.PropertyImpl;
+import org.sonar.iac.arm.tree.impl.bicep.ResourceDeclarationImpl;
+import org.sonar.iac.arm.tree.impl.bicep.StringCompleteImpl;
+import org.sonar.iac.arm.tree.impl.bicep.StringLiteralImpl;
+import org.sonar.iac.arm.tree.impl.bicep.TargetScopeDeclarationImpl;
+import org.sonar.iac.arm.tree.impl.bicep.TypeDeclarationImpl;
+import org.sonar.iac.arm.tree.impl.bicep.importdecl.ImportAsClause;
+import org.sonar.iac.arm.tree.impl.bicep.importdecl.ImportWithClause;
+import org.sonar.iac.arm.tree.impl.bicep.variable.LocalVariableImpl;
+import org.sonar.iac.common.yaml.visitors.YamlHighlightingVisitor;
+
+import static org.sonar.api.batch.sensor.highlighting.TypeOfText.ANNOTATION;
+import static org.sonar.api.batch.sensor.highlighting.TypeOfText.CONSTANT;
+import static org.sonar.api.batch.sensor.highlighting.TypeOfText.KEYWORD;
+import static org.sonar.api.batch.sensor.highlighting.TypeOfText.KEYWORD_LIGHT;
+import static org.sonar.api.batch.sensor.highlighting.TypeOfText.STRING;
+
+public class ArmHighlightingVisitor extends YamlHighlightingVisitor {
+  @Override
+  protected void languageSpecificHighlighting() {
+    register(ResourceDeclarationImpl.class, (ctx, tree) -> {
+      highlight(tree.keyword(), KEYWORD);
+      highlight(tree.symbolicName(), KEYWORD_LIGHT);
+      highlight(tree.getExisting(), KEYWORD);
+    });
+    register(DecoratorImpl.class, (ctx, tree) -> {
+      highlight(tree.keyword(), ANNOTATION);
+      Expression expression = tree.expression();
+      if (expression instanceof FunctionCall) {
+        highlight(((FunctionCall) expression).name(), ANNOTATION);
+      }
+    });
+    register(TargetScopeDeclarationImpl.class, (ctx, tree) -> highlight(tree.keyword(), KEYWORD));
+    register(ImportDeclarationImpl.class, (ctx, tree) -> highlight(tree.keyword(), KEYWORD));
+    register(ImportWithClause.class, (ctx, tree) -> highlight(tree.keyword(), KEYWORD));
+    register(ImportAsClause.class, (ctx, tree) -> highlight(tree.keyword(), KEYWORD));
```

Comment:
With a common interface, you could get rid of all the boilerplate and repetition here.